code/modules/recycling/conveyor2.dm

```diff
@@ -184,13 +211,15 @@
 	speed_process = 1
 
 
-/obj/machinery/conveyor_switch/New()
-	..()
+obj/machinery/conveyor_switch/New(newloc, newid)
+	..(newloc)
+	if(!id)
+		id = newid
 	update()
 
 	spawn(5)		// allow map load
 		conveyors = list()
-		for(var/obj/machinery/conveyor/C in world)
+		for(var/obj/machinery/conveyor/C in machines)

```

Comment:
Yeah, this definitely belongs in `initialize` - though since right now, `machines` is populated in `initialize`, you may want to move the `machines += src` in `AddAtProcessing` to `New` in <a href=https://github.com/ParadiseSS13/Paradise/blob/master/code/game/machinery/machinery.dm>here</a>, so that that list will be ready when the time comes